### PR TITLE
AR localization and tag detection

### PR DESCRIPTION
**Pull request by jeremycryan**
Developed tools to utilize April tags with the Tango.

**ar_find.py** includes methods for locating and remembering positions of encountered April tags; playing different sound samples for each tag within the camera frame; and creating a series of waypoints relative to a tag.

This pull request also contains a launch file and python file to republish April tag ID's and poses to ros.

These were tested with the Tango.

### CODE REVIEW

File: mobility_games/auditory/audio_controller.py [hunk 1]
```diff
@@ -87,17 +87,17 @@ def arrays_to_sound(list_of_tracks, quarter_time, amps = []):
...
     rate = 44100 # Sampling rate, in samples/second
     s, Hz = sHz(rate) # Seconds and hertz
     ms = 1e-3 * s
-    if synth == sin:
+    if synth == "sin":
         sound = saw_table(freq * Hz) * fadeout(fade*s)
-    elif synth == digitar:
-        sound = karplus_strong(freq*Hz * fadeout(fade*s))
+    elif synth == "digitar":
+        sound = karplus_strong(freq*Hz)# * fadeout(fade*s))
```
Comment (youralien) [inline mobility_games/auditory/audio_controller.py]: Is the comment here temporary?  What does this fadeout do?

File: mobility_games/auditory/audio_controller.py [hunk 2]
```diff
@@ -87,17 +87,17 @@ def arrays_to_sound(list_of_tracks, quarter_time, amps = []):
...
     else:
-        sound = karplus_strong(freq*Hz * fadeout(fade*s))
+        sound = karplus_strong(freq*Hz)# * fadeout(fade*s))
```
Comment (youralien) [inline mobility_games/auditory/audio_controller.py]: Here too. Comment temporarY?

File: mobility_games/games/ar_find.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+        self.yaw = angles[2]
+
+    def tag_array(self, msg):
+        # Processes the april tags currently in Tango's view.
+        # Adds a four-integer tuple to self.tag_list for each tag detected,
+        # which includes x, y, and z coordinates and the tag id.
+        self.tag_list = []
+        self.orientation_list = []
+        self.id_list = []
+        try:
```
Comment (youralien) [inline mobility_games/games/ar_find.py]: Mind commenting why you needed this try/except statement?  What error should we expect?

File: mobility_games/games/ar_find.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        print("Searching for Tango...")
+        while not rospy.is_shutdown():
+            if self.x and not self.start:
+                self.start = True
+                print("Connection established.")
+                print("Searching for tags...")
+            a = []
+            list_of_visible_tags = [cell[3] for cell in self.tag_list]
+            if len(self.tag_list) and self.start:
+                if not self.first_tag and not None in [item[1] for item in self.tag_list]:
```
Comment (youralien) [inline mobility_games/games/ar_find.py]: Item[1] is the y position right?  Why do you check to make sure the Y is not None?  Or is the selection of item[1]  arbitrary?

Comment (youralien) [inline mobility_games/games/ar_find.py]: Mind also rewriting 
`if not self.first_tag and not None in [item[1] for item in self.tag_list]` 

to

`if not self.first_tag and not (None in [item[1] for item in self.tag_list]):`

File: mobility_sensing/scripts/ar_localization.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    def __init__(self):
+        rospy.init_node('ar_localization')
+        rospy.Subscriber('/fisheye_undistorted/tag_detections_pose',
+                         PoseArray, self.republish_cane_tip)
+
+        self.pub = rospy.Publisher('/other_tag', PoseStamped, queue_size=1)
+
+        self.tag_distance = rospy.get_param('~tag_distance')
+
+    def republish_cane_tip(self, tag_msg):
```
Comment (youralien) [inline mobility_sensing/scripts/ar_localization.py]: rename, since it's less about cane tips now...

File: mobility_games/games/keypress.py [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import tty
```
Comment (youralien) [inline mobility_games/games/keypress.py]: remove keypress file, its a test file

File: mobility_games/games/frames.gv [hunk 7]
```diff
@@ -0,0 +1,14 @@
+digraph G {
```
Comment (youralien) [inline mobility_games/games/frames.gv]: add `*.gv` to the gitignore

**Review by youralien — CHANGES_REQUESTED** [review]
Just remove the unncessary files, and I'll be good to approve.  The other comments are coming more from my curiosity about the code.  Comment if you think it could be useful.  I'll fix ar_localization

File: mobility_games/auditory/audio_controller.py (hunk 1 above)
Comment (jeremycryan) [reply mobility_games/auditory/audio_controller.py]: Addressed with a comment in the code.

File: mobility_games/games/ar_find.py (hunk 3 above)
Comment (jeremycryan) [reply mobility_games/games/ar_find.py]: Addressed with comment.

File: mobility_games/games/ar_find.py (hunk 4 above)
Comment (jeremycryan) [reply mobility_games/games/ar_find.py]: The use of Y is arbitrary. Should I address this with a comment? Would it be more intuitive to use index 0?

Comment (jeremycryan) [reply mobility_games/games/ar_find.py]: Added parentheses.

**Review by youralien — APPROVED** [review]
:shipit: ship it!

**Pull request merged**